crates/chain/src/chain.rs

```diff
@@ -468,7 +468,11 @@ impl Chain {
         };
         let mut tree = db.account_state_tree()?;
         // process transactions
-        let result = match self.generator.apply_state_transition(&mut tree, args) {
+        let result = match self.generator.apply_state_transition(
+            &self.store.begin_transaction(),
```

Comment:
We should just pass the `db` to function, to make sure the uncommitted data is readable.